src/Common/tests/Tests/System/StringTests.cs

```diff
@@ -1019,9 +1019,9 @@ public static void CompareToUnknownComparisonType_StringComparison()
             Assert.Throws<ArgumentException>(() => string.Compare(value, value, (StringComparison)6));
             
             ReadOnlySpan<char> span = value.AsSpan();
-            MemoryTestHelpers.AssertThrows<ArgumentException, char>(span, (_span) => _span.CompareTo(_span, StringComparison.CurrentCulture - 1));
-            MemoryTestHelpers.AssertThrows<ArgumentException, char>(span, (_span) => _span.CompareTo(_span, StringComparison.OrdinalIgnoreCase + 1));
-            MemoryTestHelpers.AssertThrows<ArgumentException, char>(span, (_span) => _span.CompareTo(_span, (StringComparison)6));
+            SpanTestHelpers.AssertThrows<ArgumentException, char>(span, (_span) => _span.CompareTo(_span, StringComparison.CurrentCulture - 1));
+            SpanTestHelpers.AssertThrows<ArgumentException, char>(span, (_span) => _span.CompareTo(_span, StringComparison.OrdinalIgnoreCase + 1));
+            SpanTestHelpers.AssertThrows<ArgumentException, char>(span, (_span) => _span.CompareTo(_span, (StringComparison)6));
```

Comment:
Why do we need a custom helper here, rather than just doing e.g.
```C#
Assert.Throws<ArgumentException>(() => value.AsSpan().CompareTo(_span, StringComparison.CurrentCulture - 1);
```
?  I understand this will end up doing the "wrong thing" if value.AsSpan() throws an ArgumentException, but we have separate tests for that.  This would similarly fail as written if StringComparison.CurrentCulture threw an ArgumentException for example, so it's already not bullet-proof.